### ISSUE TITLE
请问怎么获取训练进度

### ISSUE THREAD

**Issue by kola515** [ultralytics/yolov5#11547]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

请问怎么获取yolov5训练过程中当前进度，或者怎么获取模型当前已经训练完成？

**Comment by glenn-jocher** [comment]
@kola515 hello! To get the training progress of YOLOv5, you can set up Tensorboard and monitor the training in real-time. To do this, you need to add the "--logdir" argument to the training command and set it to a path where you want to save the training logs. Once you start training, open Tensorboard in a web browser and point it to the saved log directory path. You should be able to see real-time training metrics like loss, mAP, etc.

Regarding your second question, to check the model's current training status you can monitor the training logs, which will indicate how many epochs have been trained so far. Alternatively, you can interrupt the training process to check the latest weights file in the "runs/train/exp/weights" directory.

I hope this helps! Let me know if you have any further questions.

**Comment by kola515** [comment]
thank you

**Comment by glenn-jocher** [comment]
Hello @kola515,

Thank you for using YOLOv5!

To detach the output tensors from the computation graph in PyTorch, you can use the `.detach()` method. This way, the output tensors will not require gradients and will not be used in the backward pass while computing gradients.

Here's an example:

```
with torch.no_grad():
    pred = model(imgs)
    
output = pred.sigmoid().detach().cpu().numpy()
```

I hope this helps! Let me know if you have any further questions.

**Comment by kola515** [comment]
>OK, thank you!

**Comment by glenn-jocher** [comment]
@kola515 you're welcome! If you have any further questions or concerns, please don't hesitate to reach out. We're here to help.
